evm/src/executor/inspector/cheatcodes/expect.rs

```diff
@@ -45,6 +45,16 @@ fn expect_revert(state: &mut Cheatcodes, reason: Option<Bytes>, depth: u64) -> R
     Ok(Bytes::new())
 }
 
+fn expect_emit(
```

Comment:
yeh my objective over time is to make handling cheatcodes a bit more state-machine like, with the ability to compose error messages and add hints and whatnot. This is a small part of it (breaking everything down into functions instead of inlining everything)